pkg/middlewares/ratelimiter/redis_limiter.go

```diff
@@ -0,0 +1,117 @@
+package ratelimiter
+
+import (
+	"context"
+	"fmt"
+	"math"
+	"net/http"
+	"time"
+
+	"github.com/redis/go-redis/v9"
+	"github.com/rs/zerolog"
+	"github.com/rs/zerolog/log"
+	ptypes "github.com/traefik/paerser/types"
+	"github.com/traefik/traefik/v3/pkg/config/dynamic"
+	"github.com/traefik/traefik/v3/pkg/middlewares/ratelimiter/redisrate"
+	"golang.org/x/time/rate"
+)
+
+type RedisLimiter struct {
+	rate     rate.Limit // reqs/s
+	burst    int64
+	maxDelay time.Duration
+
+	period  ptypes.Duration
+	logger  *zerolog.Logger
+	limiter *redisrate.Limiter
+}
+
+func NewRedisLimiter(
+	rate rate.Limit,
+	burst int64,
+	maxDelay time.Duration,
+	ttl int,
+	config dynamic.RateLimit,
+	logger *zerolog.Logger,
+) (Limiter, error) {
+	options := &redis.UniversalOptions{
+		Addrs:        config.Redis.Endpoints,
+		Username:     config.Redis.Username,
+		Password:     config.Redis.Password,
+		DB:           config.Redis.DB,
+		PoolSize:     config.Redis.PoolSize,
+		MinIdleConns: config.Redis.MinIdleConns,
+		ReadTimeout:  config.Redis.ReadTimeout,
+		WriteTimeout: config.Redis.WriteTimeout,
+		DialTimeout:  config.Redis.DialTimeout,
+	}
+	if config.Redis.TLS != nil {
+		tlsConfig, err := config.Redis.TLS.CreateTLSConfig(context.Background())
+		if err != nil {
+			return nil, err
+		}
+		options.TLSConfig = tlsConfig
+	}
+	rdb := redis.NewUniversalClient(options)
+
+	limiter := redisrate.NewLimiter(rdb, ttl, maxDelay)
```

Comment:
Instead of splitting this limiter in 2 (this struct and the `redisrate` package), did you consider having just one struct to do all this?